app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/BrowserWidget.java

```diff
@@ -146,15 +167,45 @@ public void setSurfaceTexture(SurfaceTexture aTexture, final int aWidth, final i
         } else {
             Log.e(LOGTAG, "GeckoDisplay was not null in BrowserWidget.setSurfaceTexture()");
         }
-        mDisplay.surfaceChanged(mSurface, aWidth, aHeight);
+        callSurfaceChanged();
+    }
+
+    @Override
+    public void setSurface(Surface aSurface, final int aWidth, final int aHeight, Runnable aFirstDrawCallback) {
+        GeckoSession session = SessionStore.get().getSession(mSessionId);
+        if (session == null) {
+            return;
+        }
+        mWidth = aWidth;
+        mHeight = aHeight;
+        mSurface = aSurface;
+        if (mDisplay == null) {
+            mDisplay = session.acquireDisplay();
+        } else {
+            Log.e(LOGTAG, "GeckoDisplay was not null in BrowserWidget.setSurfaceTexture()");
+        }
+        if (mSurface != null) {
+            callSurfaceChanged();
+        } else {
+            mDisplay.surfaceDestroyed();
+        }
+        if (aFirstDrawCallback != null) {
+            aFirstDrawCallback.run();
+        }
+    }
+
+    private void callSurfaceChanged() {
+        mDisplay.surfaceChanged(mSurface, mBorderWidth, mBorderWidth, mWidth - mBorderWidth * 2, mHeight - mBorderWidth * 2);
```

Comment:
I wonder if we should be making the surface larger for the border? Or are you requesting a size larger than you want the content size to be? Basically, if we want our content to be 200x300 with a 1 pixel border, we should be requesting a surface size of 202x302.